nifi-commons/nifi-vault-utils/src/main/java/org/apache/nifi/vault/hashicorp/HashiCorpVaultCommunicationService.java

```diff
@@ -24,6 +25,10 @@
  * @see <a href="https://www.vaultproject.io/">https://www.vaultproject.io/</a>
  */
 public interface HashiCorpVaultCommunicationService {
+    /**
+     * Tests the connection to Vault, throwing an exception if there was a problem connecting to the server.
+     */
+    void testConnection();
```

Comment:
Instead of having this as a void method, what about changing it to `getServerVersion()`, and return the `String` from `VaultHealth`? That would make it a little more useful, and serve the same purpose for testing connectivity.